log4j-api/src/main/java/org/apache/logging/log4j/util/Base64Util.java

```diff
@@ -53,7 +54,7 @@ public static String encode(final String str) {
         if (str == null) {
             return null;
         }
-        final byte [] data = str.getBytes();
+        final byte [] data = str.getBytes(Charset.defaultCharset());
```

Comment:
Shouldn't rather we getting the charset from the caller? _Default charset_ is an assumption and can very well be wrong.

Reply:
I marked the method as `@Deprecated` and removed its only user. The entire `Base64Util` class can be replaced with `Base64.getEncoder()` since Java 8.